src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,379 @@
+        dxdt[4]  = q * cos_phi - r * sin_phi                                    # Euler's angle theta increment
+        dxdt[5]  = q * sin_phi / cos_theta + r * cos_phi / cos_theta            # Euler's angle psi increment
+        
+        dxdt[6]  = ((sin_theta * cos_psi * cos_phi + sin_phi * sin_psi) * T - fe_drag[0]) / m   # Acceleration along x-axis
+        dxdt[7]  = ((sin_theta * sin_psi * cos_phi - sin_phi * cos_psi) * T - fe_drag[1]) / m   # Acceleration along y-axis
+        dxdt[8]  = - self.parameters['gravity'] + (cos_phi * cos_theta  * T - fe_drag[2]) / m   # Acceleration along z-axis
+
+        dxdt[9]  = ((Iyy - Izz) * q * r + tp * self.geom['arm_length']) / Ixx     # Angular acceleration along body x-axis: roll rate
+        dxdt[10] = ((Izz - Ixx) * p * r + tq * self.geom['arm_length']) / Iyy     # Angular acceleration along body y-axis: pitch rate
+        dxdt[11] = ((Ixx - Iyy) * p * q + tr *        1               ) / Izz     # Angular acceleration along body z-axis: yaw rate
+        dxdt[12] = 1                                                              # Auxiliary time variable
+        dxdt[13] = (u['mission_complete'] - x['mission_complete'])/self.parameters['dt']    # Value to keep track of percentage of mission completed
+        
+        return self.StateContainer(np.array([np.atleast_1d(item) for item in dxdt]))
+    
+    def event_state(self, x : dict) -> dict:
+
+        # Based on percentage of reference trajectory completed 
+        return {
+                x['mission_complete']
+        }
+ 
+    def output(self, x : dict):
+        # Output is the same as the state vector, without time and mission_complete
+        return self.OutputContainer(x.matrix[0:-2])
+
+    def threshold_met(self, x : dict) -> dict:
+        
+        # Progress through the reference trajectory is saved in the state 'mission_complete' 
+        if x['mission_complete'] >= 1:
+            return {
+                'TrajectoryComplete': True
+            }
+        else:
+            return {
+                'TrajectoryComplete': False 
+            }
+
+    def simulate_to_threshold(self, future_loading_eqn, first_output = None, threshold_keys = None, **kwargs):
+
+        # Check for appropriately defined dt - must be same as vehicle model 
+        if 'dt' in kwargs and kwargs['dt'] != self.parameters['dt']:
+          kwargs['dt'] = self.parameters['dt']
+          warn("Simulation dt must be equal to dt defined for the vehicle model. dt = {} is used.".format(self.parameters['dt'])) 
+        elif 'dt' not in kwargs:
+          kwargs['dt'] = self.parameters['dt']
+
+        # Simulate to threshold
+        sim_res = super().simulate_to_threshold(future_loading_eqn,first_output, threshold_keys, **kwargs)
+
+        return sim_res
+
+    def linear_model(self, phi, theta, psi, p, q, r, T):
+        """ The linear model ignores gyroscopic effect and wind rate of change"""
```

Comment:
```suggestion
        """
        Generates A, B matrices of simplified linear model. Matricies ignore gyroscopic effect and wind rate of change
        """
```
Also, could use description of arguments and returns